Source/PluginEditor.cpp

```diff
@@ -22,12 +22,16 @@ void fadePostEndRegion(int endRangeBorderX, Graphics& g);
 //==============================================================================
 SoomplerAudioProcessorEditor::SoomplerAudioProcessorEditor (SoomplerAudioProcessor& p)
     : AudioProcessorEditor (&p), processor (p), mainFont("DejaVu Sans", 12, Font::plain), startRangeX(Settings::THUMBNAIL_BOUNDS.getX()),
-      endRangeX(Settings::THUMBNAIL_BOUNDS.getRight()), pianoRoll((MidiEventSupplier&) processor, (MidiEventConsumer&) processor)
+      endRangeX(Settings::THUMBNAIL_BOUNDS.getRight()), pianoRoll((MidiEventSupplier&) processor, (MidiEventConsumer&) processor),

```

Comment:
i think there is too many concrete types. dynamic casting is indicator of that issue. Maybe you can do something with ordinary inheritance and polimorphism?

Reply:
I think this dynamic casts is obsolete now, I will delete them in next releases